File: disk-ann/src/main/java/jetbrains/exodus/diskann/bench/GIST1MBench.java
package jetbrains.exodus.diskann.bench;

import java.io.IOException;
import java.nio.file.Path;
import java.util.Objects;

public class GIST1MBench {
    public static void main(String[] args) {
        var benchPathStr = System.getProperty("bench.path");
        Path benchPath;

        benchPath = Path.of(Objects.requireNonNullElse(benchPathStr, "."));

        try {
            BenchUtils.runSiftBenchmarks(
                    benchPath, "gist", "gist.tar.gz", "gist_base.fvecs",
                    "gist_query.fvecs",
                    "gist_groundtruth.ivecs", 960
            );
        } catch (IOException e) {
            e.printStackTrace();
            throw new RuntimeException(e);
        }
    }
}

File: disk-ann/src/main/kotlin/jetbrains/exodus/diskann/bench/GIST1MBench.kt
package jetbrains.exodus.diskann.bench

import java.nio.file.Path

fun main() {
    val benchPathStr = System.getProperty("bench.path")
    val benchPath = if (benchPathStr != null) {
        Path.of(benchPathStr)
    } else {
        Path.of(".")
    }

    BenchUtils.runSiftBenchmarks(
        benchPath, "gist", "gist.tar.gz", "gist_base.fvecs",
        "gist_query.fvecs",
        "gist_groundtruth.ivecs", 960
    )
}
